.github/workflows/build.yml

```diff
@@ -0,0 +1,134 @@
+name: "Test & Build"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up MongoDB
+        uses: ankane/setup-mongodb@v1
+        with:
+          mongodb-version: '5.0'
+
+      - name: Set up JDK 17 (Temurin)
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Compile
+        run: |
+          mvn -q -U -B compile
+
+      - name: Test
+        run: |
+          mvn -q -U -B test
+
+      - name: Package
+        run: |
+          mvn -q -U -B -ff -DskipTests package
+
+      - name: Verify
+        run: |
+          mvn -q -U -B -ff -DskipTests package
```

Comment:
No strong feelings either way, splitting this in four steps. I assume the benefit is more granular reporting?